compiler-rt/lib/asan/asan_linux.cpp

```diff
@@ -114,8 +114,12 @@ void ReExecWithoutASLR() {
   // this function as a last resort (when the memory mapping is incompatible
   // and ASan would fail anyway).
   int old_personality = personality(0xffffffff);
-  bool aslr_on =
-      (old_personality != -1) && ((old_personality & ADDR_NO_RANDOMIZE) == 0);
+  if (old_personality == -1) {
+    VReport(1, "WARNING: unable to run personality check.\n");
+    return;
+  }
+
+  bool aslr_on = ((old_personality & ADDR_NO_RANDOMIZE) == 0);
```

Comment:
nit: unnecessary parens

Reply:
I removed the outermost parentheses. I left the inner parentheses for clarity because I always need a double take on the order of precedence for & vs. ==.

Reply:
I did only mean the other one, thanks.